src/tests/simulation/tests/integration_tests/scale/cordon.rs

```diff
@@ -0,0 +1,421 @@
+            }
+
+            // cordon on running cluster
+            #[madsim::test]
+            async fn [< cordoned_nodes_do_not_get_new_actors_1_ $query >]() -> Result<()> {
+                use risingwave_simulation::nexmark::queries::$query::*;
+                cordoned_nodes_do_not_get_new_actors(CREATE, SELECT, DROP, 1).await
+            }
+            #[madsim::test]
+            async fn [< cordoned_nodes_do_not_get_new_actors_2_ $query >]() -> Result<()> {
+                use risingwave_simulation::nexmark::queries::$query::*;
+                cordoned_nodes_do_not_get_new_actors(CREATE, SELECT, DROP, 2).await
+            }
+
+            // Idempotent
+            #[madsim::test]
+            async fn [< cordon_is_idempotent_ $query >]() -> Result<()> {
+                use risingwave_simulation::nexmark::queries::$query::*;
+                cordon_is_idempotent(CREATE, SELECT, DROP, 1).await
+            }
+            #[madsim::test]
+            async fn [< cordon_is_idempotent_2_ $query >]() -> Result<()> {
+                use risingwave_simulation::nexmark::queries::$query::*;
+                cordon_is_idempotent(CREATE, SELECT, DROP, 2).await
+            }
+
+            // cordon all nodes
+            #[madsim::test]
+            async fn [< cordon_all_nodes_error_ $query >]() {
+                use risingwave_simulation::nexmark::queries::$query::*;
+                let result = cordoned_nodes_do_not_get_actors(CREATE, SELECT, DROP, 3)
+                    .await;
+                assert!(result.is_err());
+                assert_eq!(result.err().unwrap().to_string(), "db error: ERROR: QueryError: internal error: Service unavailable: No available parallel units to schedule");
+            }
+            #[madsim::test]
+            async fn [< cordon_all_nodes_error_2_ $query >]() {
+                use risingwave_simulation::nexmark::queries::q3::*;
+                let result = cordoned_nodes_do_not_get_new_actors(CREATE, SELECT, DROP, 3)
+                    .await;
+                assert!(result.is_err());
+                assert_eq!(result.err().unwrap().to_string(), "db error: ERROR: QueryError: internal error: Service unavailable: No available parallel units to schedule");
+            }
+
+            // invalid scheduling request
+            #[madsim::test]
+            async fn [< invalid_reschedule_1_ $query >]() -> Result<()> {
+                use risingwave_simulation::nexmark::queries::$query::*;
+                invalid_reschedule(CREATE, SELECT, DROP, 1)
+                    .await
+            }
+            #[madsim::test]
+            async fn [< invalid_reschedule_2_ $query >]() -> Result<()> {
+                use risingwave_simulation::nexmark::queries::q3::*;
+                invalid_reschedule(CREATE, SELECT, DROP, 2)
+                    .await
+            }
+        }
+    };
+}
+
+// q0, q1, q2: too trivial
+test!(q3);
```

Comment:
I think we just need to cover a very basic test.

Reply:
What exactly do you need by a basic test? Should I remove the other queries? Should it only be `test!(q3)`? 

I would strongly recommend to keep the different tests themself. They really helped me catch issues when developing and I would assume that they will be helpful in the future as well.

Reply:
I worked 🥳 The diff is less confusing now